aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -809,6 +810,12 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
   // Keep query, key, value in high precision for accuracy
   // NestedTensor reports issues for backward with autograd so disabled: must be
   // contiguous to get buffer.
+  auto autocast_state = at::autocast::is_autocast_enabled(query_.device().type());
+  if (!ctx.allowFP16BF16ReductionMathSDP()) {
+    // If a user wants to disable FP16 and BF16 Reduction we also want to
+    // disable autocast
+    at::autocast::set_autocast_enabled(query_.device().type(), false);
+  }
```

Comment:
nit: if you end up going this route, might be better to save / restore using an RAII guard to be exception-safe